cmake/scripts/ROOTConfig.cmake.in

```diff
@@ -112,10 +112,11 @@ if (NOT ROOT_builtin_nlohmannjson_FOUND)
           set(_need_json TRUE)
        endif()
     endforeach()
-  else()
-    # if components not specified - suppose all
-    set(_need_json TRUE)
   endif()
+  # If components are not specified, we assume json is not required.
+  # Propagating the dependency on nlohmann_json just for the sake of one small
+  # ROOT feature that might be used is unreasonable. We can expect from the
+  # users that link against ROOTEve to list this component explicitly.
```

Comment:
What do you mean by that?  Who is the 'users' in this context?  Is it the developers of `Eve` or the end-users of `Eve`.  What do they now need to do that they were not doing before?

Reply:
It's the users of `Eve` that have their own CMake project.

Before, they could have done:
```CMake
   find_package(ROOT COMPONENTS ROOTEve)
```
or just
```CMake
   find_package(ROOT)
```

Now with this PR, it doesn't assume anymore that you use the `ROOTEve` component when you don't specify any components. So in that case, you'd need to find nlohmann_json yourself.

So now it's either:
Before, they could have done:
```CMake
   find_package(ROOT COMPONENTS ROOTEve)
```
or
```CMake
   find_package(ROOT)
   # we're using REve, which requires nlohmann_json:
   find_package(nlohmann_json)
```

Reply:
```   
find_package(ROOT)
# we're using REve, which requires nlohmann_json:
find_package(nlohmann_json)
```

Does this find nlohmann_json from builtin ROOT if that was builtin, or will it potentially give a spurious warning when no nlohmann_json exists on the system, and is also not needed because it comes with ROOT?

Reply:
`find_package(nlohmann_json)` will search for system-wide nlohmann_json.

Now ROOT will search for such package only if `ROOTEve` is listed in required components and if ROOT itself was build with external `nlohmann_json`.

Reply:
So the answer is yes, it will create a spurious warning. So one shouldn't do that `or` suggestion unless one is certain that nlohmann_json always comes from the system? Because library developers can't be sure under what circumstances their users obtained ROOT.

Reply:
But why one needs `find_package(nlohmann_json)`?  
Only if `nlohmann/json.hpp` used in the code. 
Otherwise such search can be skipped - isn't it?

Reply:
One needs it because ROOTEve potentially (!!!!) depends on it if external nlohmann_json was used. As a Library author I would like to not worry about that, so using `COMPONENTS ROOTEve` is the one thing that people have to do with what is here now, isn't it?

Reply:
`COMPONENTS ROOTEve` need to be listed if one really uses this library and compiles code which includes headers like `REveManager.hxx`. And this is very special applications.

In all other cases one can skip `COMPONENT` and not need care about `nlohmann_json`.

Reply:
Sorry, I thought it was clear that I do use ROOTEve in my library, so I do care.

Reply:
My point is that the second example is wrong.

Reply:
Yes, after this PR there is no clear way to use same `nlohmann/json.hpp` version as ROOT.
And example with simple `find_package(nlohmann_json)` is not correct in this respect.
Maybe one can introduce `ROOT::nlohmann_json` library which solves such problem.

Reply:
And avoid giving that example here, in case other people stumble over this issue.

Reply:
Ok, thanks for having figured this out! Then, one indeed should do `find_package(ROOT COMPONENTS ROOTEve)` as @linev suggested. I see that you did this in dd4hep now, so the problem is solved for you?

Reply:
Yeah, works for me now :)

Reply:
> Yes, after this PR there is no clear way to use same `nlohmann/json.hpp` version as ROOT. And example with simple `find_package(nlohmann_json)` is not correct in this respect. Maybe one can introduce `ROOT::nlohmann_json` library which solves such problem.

I think that's not needed, at least not if nobody asks for it. The real power users don't use builtins anyway, and if ROOT uses `nlohmann_json` from the system, then it's easy to use the same version in your software that uses ROOT.

Reply:
> And avoid giving that example here, in case other people stumble over this issue.

I have edited my previous comment, explaining that the example was wrong and one should request the `ROOTEve` component explicitly.

Reply:
> I think that's not needed, at least not if nobody asks for it. The real power users don't use builtins anyway, and if ROOT uses nlohmann_json from the system, then it's easy to use the same version in your software that uses ROOT.

If I remember correctly - there was ROOT build using `nlohmann/json.hpp` from custom sub-directory. At the same time in `/usr/include` older version of `nlohmann/json.hpp` was installed.  And without command (which is removed now by PR) this leads to building user code against two different versions of `nlohmann/json.hpp`. If I am right - this was failing when trying to create dictionary.

The only solution after this PR - do `find_package(ROOT COMPONENTS ROOTEve)` even when `ROOTEve` is not used.